src/Internal/QueuedWritesFile.php

```diff
@@ -38,7 +38,7 @@ public function __construct(
         }
 
         $this->queue = new \SplQueue();
-        $this->writable = $this->mode[0] !== 'r';
+        $this->writable = $this->mode[0] !== 'r' || $this->mode == 'r+';
```

Comment:
This likely needs normalization like in other places and can then just check for `$this->mode !== 'r'`?

See `$mode = \str_replace(['b', 't', 'e'], '', $mode);` in other places.

Reply:
https://github.com/amphp/file/blob/efeb737b1f19a7fcfe8a763c5ef2ad80dbac93dd/src/functions.php#L78

https://github.com/danog/MadelineProto/blob/0689c16309a8a404f920f5e5543b64ad8a3309b2/src/MTProtoTools/FilesLogic.php#L360

The openFile is amphp function. Maybe I am wrong, but AS I can understand, so IF it possible to create new file with amphp library with "r+" permission, so the library will do the rest -> create it, write it, ...

If I am wrong could you please explain to me. Thanks!

Reply:
@yurganov I think you didn't understand correctly: The new check is better than the old one, but still potentially incorrect, e.g. for `rb+` (at least I think that works). So it should normalize the mode with the mentioned function and then check just for `!== 'r'`, as that's the only mode that is non-writable.

Reply:
@kelunik Yeah, sure, I forgot about rb. I think it's easier to validate only r and rb.

Reply:
@yurganov I still think we should remove `b`, `t`, and `e` from the mode, as they're just flags and that can be present in any order IIRC.